### PR TITLE
#214 Refactor db connect methods to individual plugin methods

### PR DESCRIPTION
**Pull request by sworup**

### CODE REVIEW

**Comment by andreievg** [comment]
@CarlosNZ, would you mind looking at this please ?

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Great, thanks for making a start on these. I've made a couple of small comments in the code.

Also:

- Can you merge the latest `master` into this? I'd like to check that all the tests work before approving, and there's some that are failing now and were only fixed in a later PR to the one you've branched off
- We seem to be having a problem with the compiled .js files on plugins that have multiple .js files in them (See issue #221). For now, can you leave out the .js files on any of the plugins you've worked on (just delete them) until we figure out what's going on there

File: src/plugins/types.ts [hunk 1]
```diff
@@ -6,6 +6,12 @@ export interface ActionPluginOutput {
   output?: { [key: string]: any }
 }
 
+export enum ApplicationOutcome {
+  Pending = 'PENDING',
+  Approved = 'APPROVED',
+  Rejected = 'REJECTED',
+}
+
```
[the remark this answers is not in the document]
Comment (sworup) [reply src/plugins/types.ts]: Resolved with 8c17f42

**Comment by CarlosNZ** [comment]
Thanks @sworup, can you just delete the extra .js files, cos we're still getting the error I mentioned above? Thx

**Comment by andreievg** [comment]
hmm, i think there are too many conflicts now, and we've decided to just relocate the methods when an action is touched. closing

**Pull request closed without merging**